tests/models/llm_embed/test_llm_embedding.py

```diff
@@ -0,0 +1,451 @@
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.config: AutoConfig = AutoConfig.from_pretrained(
+            'bert-base-uncased',
+        )
+        self.config.hidden_size = 768
+        self.config.num_hidden_layers = 12
+        self.config.n_layers = 12
+        self.config.vocab_size = 30000
+        self.linear: torch.nn.Linear = torch.nn.Linear(
+            self.config.hidden_size,
+            self.config.hidden_size,
+        )
+
+    @classmethod
+    def from_pretrained(cls, *args: Any, **kwargs: Any) -> 'MockAutoModel':
+        return cls()
+
+    def forward(
+        self,
+        **kwargs: Any,
+    ) -> BaseModelOutputWithPastAndCrossAttentions:
+        # Simulate forward pass
+        input_ids: torch.Tensor = kwargs.get(
+            'input_ids',
+            torch.zeros(1, 10, dtype=torch.long),
+        )
+        batch_size: int = input_ids.size(0)
+        seq_length: int = input_ids.size(1)
+        last_hidden_state: torch.Tensor = torch.randn(
+            batch_size,
+            seq_length,
+            self.config.hidden_size,
+        )
+        last_hidden_state = self.linear(last_hidden_state)
+        return BaseModelOutputWithPastAndCrossAttentions(
+            last_hidden_state=last_hidden_state,
+            hidden_states=(last_hidden_state,) *
+            (self.config.num_hidden_layers + 1),
+        )
+
+
+@pytest.fixture
+def mock_auto_model() -> MockAutoModel:
+    return MockAutoModel()
+
+
+@pytest.fixture
+def model(
+    mock_tokenizer: MockTokenizer,
+    mock_auto_model: MockAutoModel,
+) -> ContrastiveModel:
+    with patch('transformers.AutoModel.from_pretrained', return_value=mock_auto_model), \
+         patch('torch.distributed.is_initialized', return_value=False), \
+         patch('torch.distributed.get_rank', return_value=0), \
+         patch('torch.distributed.barrier'), \
+         patch('llmfoundry.models.llm_embed.FinetuneEmbeddingModel.construct_model', return_value=mock_auto_model):
+        model_instance: ContrastiveModel = ContrastiveModel(
+            tokenizer=mock_tokenizer,
+            pretrained_model_name_or_path='bert-base-uncased',
+            loss_fn='torch_crossentropy',
+            use_flash_attention_2=False,
+        )
+        return model_instance
+
+
+def build_lm_config(is_hf: bool, attn_impl: Optional[str]) -> dict[str, Any]:
+    if is_hf:
+        assert attn_impl is None
+        return {'pretrained_model_name_or_path': 'facebook/opt-350m'}
+    else:
+        assert attn_impl is not None
+        return {
+            # bare config
+            'num_layers': 2,
+            'd_model': 64,
+            'vocab_size': 100352,
+            'attn_config': {
+                'attn_impl': attn_impl,
+            },
+        }
+
+
+def build_tokenizer_config(is_hf: bool):
+    if is_hf:
+        return {'name': 'facebook/opt-350m', 'kwargs': {}}
+    else:
+        return {'name': 'tiktoken', 'kwargs': {'model_name': 'gpt-4'}}
```

Comment:
you can use the same tokenizer in both cases, just change vocab size accordingly